internal/controllers/topology/cluster/patches/engine.go

```diff
@@ -193,7 +193,7 @@ func addVariablesForPatch(blueprint *scope.ClusterBlueprint, desired *scope.Clus
 		if item.HolderReference.Kind == "MachineDeployment" {
 			md, ok := mdStateIndex[item.HolderReference.Name]
 			if !ok {
-				return errors.Errorf("could not find desired state for MachineDeployment %s", klog.KObj(md.Object))
+				return errors.Errorf("could not find desired state for MachineDeployment %s", klog.KRef(item.HolderReference.Namespace, item.HolderReference.Name))
```

Comment:
This panic'ed previously if actually hit this line

Reply:
Nice catch!